### PR TITLE
[#5745] feat(CLI): Table format output for ListCatalogs command

### PR DESCRIPTION
**Pull request by waukin**
### What changes were proposed in this pull request?

Support table format output for ListCatalogs command.

### Why are the changes needed?

Issue: https://github.com/apache/gravitino/issues/5745

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

```
gcli catalog list -m <metalake_name>
gcli catalog list -m <metalake_name> --output plain
gcli catalog list -m <metalake_name> --output table
```

### CODE REVIEW

**Comment by waukin** [comment]
Hi @xunliu, This PR is ready for review. Please take a look.

File: clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java [hunk 1]
```diff
@@ -33,6 +33,8 @@ public static void output(Object object) {
       new MetalakesStringFormat().output((Metalake[]) object);
     } else if (object instanceof Catalog) {
       new CatalogStringFormat().output((Catalog) object);
+    } else if (object instanceof Catalog[]) {
+      new CatalogsStringFormat().output((Catalog[]) object);
```
Comment (xunliu) [inline clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java:37]: Please change `CatalogsStringFormat` to `CatalogsPlainFormat`,  and `MetalakesStringFormat `, `CatalogStringFormat ` and other `*StringFormat` class

File: clients/cli/src/main/java/org/apache/gravitino/cli/outputs/TableFormat.java [hunk 2]
```diff
@@ -79,6 +81,19 @@ public void output(Catalog catalog) {
     }
   }
 
+  static final class CatalogsTableFormat implements OutputFormat<Catalog[]> {
+    @Override
+    public void output(Catalog[] catalogs) {
+      List<String> headers = Collections.singletonList("catalog");
+      List<List<String>> rows = new ArrayList<>();
+      for (int i = 0; i < catalogs.length; i++) {
+        rows.add(Arrays.asList(catalogs[i].name()));
```
Comment (xunliu) [inline clients/cli/src/main/java/org/apache/gravitino/cli/outputs/TableFormat.java:90]: I think we can display more filed values, just like `type()`, `provider()`, `comment` and `properties()`.

Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/outputs/TableFormat.java:90]: That is what would be displayed in a details command not a list command.

File: clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java (hunk 1 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java:37]: I wodl call them `CatalogStringFormat`, `CatalogPlainFormat` etc

Comment (xunliu) [reply clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java:37]: The `CatalogPlainFormat` output `Catalog` object.
The `CatalogsPlainFormat` output `Catalog[]` objects.
...

File: clients/cli/src/main/java/org/apache/gravitino/cli/outputs/TableFormat.java (hunk 2 above)
Comment (xunliu) [reply clients/cli/src/main/java/org/apache/gravitino/cli/outputs/TableFormat.java:90]: I agreed.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 3]
```diff
@@ -201,7 +201,7 @@ private void handleCatalogCommand() {
     String outputFormat = line.getOptionValue(GravitinoOptions.OUTPUT);
 
     if (CommandActions.LIST.equals(command)) {
-      newListCatalogs(url, ignore, metalake).handle();
+      newListCatalogs(url, ignore, metalake, outputFormat).handle();
```
Comment (tengqm) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:204]: I'd suggest we come up with a ListOptions wrapper for this case.
In the foreseeable future, you may want to add filter, sorting, field selector, paging options. The list of parameters to pass would grow and you have to change all these places again and again. This process is tedious and error prone.

Comment (xunliu) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:204]: I think we can pass the CommondOption value in the function params.
Through `CommondOption` we can get any option values.
@tengqm  @justinmclean What do you think?

Comment (tengqm) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:204]: I'd suggest we don't overly generalize the options.
For example, you may want to model `ListOptions`, `DeleteOptions`, `UpdateOptions`.
The `outputFormat`, or `sortBy` option can then be encapsulated into `ListOptions`,
while an option like `force` can be encapsulated into `DeleteOptions`.

Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:204]: I would not pass the CommondOption in the function parameters, so the command needs to know about the CLI library we use, and it makes it hard to know what options a command uses. Currently, we only have one list option and one delete option. I think we could change to a ListOptions or similar when adding extra options.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 4]
```diff
@@ -208,11 +209,12 @@ private void handleCatalogCommand() {
     FullName name = new FullName(line);
     String metalake = name.getMetalakeName();
     String outputFormat = line.getOptionValue(GravitinoOptions.OUTPUT);
+    ListOptions listOptions = new ListOptions(outputFormat);
 
     Command.setAuthenticationMode(auth, userName);
 
     if (CommandActions.LIST.equals(command)) {
-      newListCatalogs(url, ignore, metalake).handle();
+      newListCatalogs(url, ignore, metalake, listOptions).handle();
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:217]: The order of the parameters should be the same, other calls have listOptions before metlake.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 5]
```diff
@@ -159,17 +159,18 @@ private void handleMetalakeCommand() {
     FullName name = new FullName(line);
     String metalake = name.getMetalakeName();
     String outputFormat = line.getOptionValue(GravitinoOptions.OUTPUT);
+    ListOptions listOptions = new ListOptions(outputFormat);
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:162]: This doesn't currently add any value to the code and makes it harder to understand. ListOption is also a poor name as it is used in both list and details commands.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/CatalogDetails.java [hunk 6]
```diff
@@ -35,14 +36,17 @@ public class CatalogDetails extends Command {
    *
    * @param url The URL of the Gravitino server.
    * @param ignoreVersions If true don't check the client/server versions match.
-   * @param outputFormat The output format.
+   * @param listOptions The list options.
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/CatalogDetails.java:39]: The name of this variable is confusing.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/Command.java [hunk 7]
```diff
@@ -168,14 +169,14 @@ protected <T extends GravitinoClientBase> Builder<T> constructClient(Builder<T>
    * @param <T> The type of entity.
    */
   protected <T> void output(T entity) {
-    if (outputFormat == null) {
+    if (listOptions.getOutputFormat() == null) {
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/Command.java:172]: Can listOptions be null?

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListCatalogs.java [hunk 8]
```diff
@@ -35,29 +36,26 @@ public class ListCatalogs extends Command {
...
   /** Lists all catalogs in a metalake. */
   @Override
   public void handle() {
-    String[] catalogs = new String[0];
+    Catalog[] catalogs;
     try {
       GravitinoClient client = buildClient(metalake);
-      catalogs = client.listCatalogs();
+      catalogs = client.listCatalogsInfo();
+      output(catalogs);
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListCatalogs.java:54]: This command is a list command that only lists the catalog names. The details command lists more information. This is now inconsistent with other commands. Why was this changed?

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java (hunk 5 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:162]: displayOptions might be a better name? But as I previously commented we should only add this after we have more than one display option.

Comment (waukin) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:162]: So you mean we don't need a `ListOptions` wrapper at this stage because we currently have only one display option, namely outputFormat?

Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:162]: I think so, it adds extra complexity and decreases code readability for little or no benefit. I think it would be useful if we had 2 or 3 options.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/CatalogDetails.java (hunk 6 above)
Comment (waukin) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/CatalogDetails.java]: Reverted to using outputFormat for @param.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java (hunk 4 above)
Comment (waukin) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java]: Update to `newListCatalogs(url, ignore, outputFormat, metalake).handle();`

File: clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java [hunk 9]
```diff
@@ -145,21 +146,19 @@ public void testMetalakeDetailsCommand() {
   }
 
   @Test
-  public void testCatalogDetailsCommand() {
+  public void testCatalogListCommand() {
```
Comment (justinmclean) [inline clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java:149]: I'm not sure why this has changed from a Details command to a List command don't we want to test both?

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListCatalogs.java (hunk 8 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListCatalogs.java:52]: OK I understand

File: clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java (hunk 9 above)
Comment (waukin) [reply clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java:149]: Sorry, I lost the `testCatalogDetailsCommand()` during the rebase, but I have already reverted it.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/Command.java (hunk 7 above)
Comment (waukin) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/Command.java]: The `listOptions` feature is not implemented at this stage.

**Review by xunliu — APPROVED** [review]
@waukin Thank you for your contributions.
LGTM
